spikeinterface/curation/auto_merge.py

```diff
@@ -311,24 +315,59 @@ def compute_templates_diff(sorting, templates, num_channels=5, num_shift=5, pair
 
     return templates_diff
 
+class MockWaveformExtractor:
+    def __init__(self, recording, sorting):
+        self.recording = recording
+        self.sorting = sorting
 
-def check_improve_contaminations_score(sorting, pair_mask, contaminations):
+def check_improve_contaminations_score(we, pair_mask, contaminations,
+        firing_contamination_balance, refractory_period_ms, censored_period_ms):
     """
+    Check that the score is improve afeter a potential merge
+
+    The score is a balance between:
+      * contamination decrease
+      * firing increase
+
     Check that the contamination score is improved (decrease)  after 
     a potential merge
     """
+    recording = we.recording
+    soring = we.sorting
     pair_mask = pair_mask.copy()
 
+    
+
+    firing_rates = compute_firing_rate(we)
+
     inds1, inds2 = np.nonzero(pair_mask)
     for i in range(inds1.size):
         ind1, ind2 =inds1[i], inds2[i]
 
-        c1 = contaminations[ind1]
-        c2 = contaminations[ind2]
-
-        
-
+        c_1 = contaminations[ind1]
+        c_2 = contaminations[ind2]
 
+        f_1 = firing_rates[ind1]
+        f_2 = firing_rates[ind2]
 
+        # make a merged sorting and tale one unit (unit_id1 is used)
+        unit_id1, unit_id2 = sorting.unit_ids[ind1], sorting.unit_ids[ind2]
+        sorting_merged = MergeUnitsSorting(sorting, [unit_id1, unit_id2], new_unit_id=unit_id1).select_units([unit_id1])
+        # make a lazy fake WaveformExtractor to compute contamination and firing rate
```

Comment:
Once we merge `remove_duplicated_spikes`, we need to add it here.

Reply:
Yep but this should beequivalent normally.